src/main.rs

```diff
@@ -203,36 +175,41 @@ mod app {
         }
     }
 
-    #[task(priority = 1, local=[dac, pwm], shared=[settings, gpio], capacity=1)]
-    fn settings_update(mut c: settings_update::Context, settings: Settings) {
+    #[task(priority = 1, local=[pwm], shared=[dac, settings, gpio], capacity=1)]
+    fn settings_update(mut c: settings_update::Context, mut settings: Settings) {
+        // Limit y_min and y_max values here. Will be incorporated into miniconf response later.
+        settings.output_channel.iter_mut().for_each(|ch| {
+            ch.iir.y_max = ch.iir.y_max.clamp(0.0, DacCode::MAX_CURRENT as f64);
+            ch.iir.y_min = ch.iir.y_min.clamp(-3.0, 0.0);
```

Comment:
Why the -3.0, why the 0.0?

```suggestion
            ch.iir.y_max = ch.iir.y_max.clamp(-DacCode::MAX_CURRENT as _, DacCode::MAX_CURRENT as _);
            ch.iir.y_min = ch.iir.y_min.clamp(-DacCode::MAX_CURRENT as _, DacCode::MAX_CURRENT as _);
```

Reply:
true, that makes no sense to limit this here